lib/stat_generator.rb

```diff
@@ -71,6 +71,69 @@ def id_by_season
     season_id_list
   end
 
+  def stats_by_id_and_season
```

Comment:
Really eager to know if there is a better way, but man, well done. It can't have been easy getting some of these methods down.